src/main/java/edu/ie3/datamodel/io/source/csv/CsvWeatherSource.java

```diff
@@ -205,10 +205,7 @@ private Try<Stream<Map<String, String>>, SourceException> buildStreamWithFieldsT
       // is wanted to avoid a lock on the file), but this causes a closing of the stream as well.
       // As we still want to consume the data at other places, we start a new stream instead of
       // returning the original one
-      Collection<Map<String, String>> allRows =
-          dataSource.csvRowFieldValueMapping(reader, headline);
-
-      return Success.of(dataSource.checkExactDuplicates("Weather", allRows).parallelStream());
+      return Success.of(dataSource.csvRowFieldValueMapping(reader, headline).parallelStream());
```

Comment:
I mean the way you implemented, comparing only the unique columns.

Reply:
Yes, sorry for the confusion. It shouldn't be a big change, just applying the methods you created, if possible

Reply:
The question is, do I apply the method to the build data or to the raw strings?

Reply:
You mean applying to the build data?

Reply:
Yes, so in `CsvWeatherSource` this would happen somewhere towards the end of `readWeatherTimeSeries`. Or are there arguments against including the uniqueness validation?